### PR TITLE
CredentialComposer facade and catalog support

### PR DESCRIPTION
**Pull request by azdagron**
This change supplies a facade for the v1 CredentialComposer plugin interface.

It also provides the catalog scaffolding but does not yet have the catalog load CredentialComposer plugins until the rest of the supporting code is in place in future changes.

This is the first of a set of PRs to introduce the feature tracked by https://github.com/spiffe/spire/issues/3253.

### CODE REVIEW

File: pkg/server/plugin/credentialcomposer/v1_test.go [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+				SpiffeId: id.String(),
+				Attributes: &credentialcomposerv1.JWTSVIDAttributes{
+					Claims: &structpb.Struct{Fields: map[string]*structpb.Value{"ORIGINAL_KEY": structpb.NewStringValue("ORIGINAL_VALUE")}},
+				},
+			},
+			responseOut:         &credentialcomposerv1.ComposeWorkloadJWTSVIDResponse{},
+			expectAttributesOut: credentialcomposer.JWTSVIDAttributes{Claims: map[string]interface{}{"ORIGINAL_KEY": "ORIGINAL_VALUE"}},
+		},
+		{
+			test:         "attributes unchanged if plugin does not respond with attributes",
```
Comment (amartinezfayo) [inline pkg/server/plugin/credentialcomposer/v1_test.go:862]: Test description needs to be updated here.

File: pkg/server/plugin/credentialcomposer/v1.go [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+	if err != nil {
+		return pkix.Name{}, fmt.Errorf("extra names: %w", err)
+	}
+	return pkix.Name{
+		Country:            in.Country,
+		Organization:       in.Organization,
+		OrganizationalUnit: in.OrganizationalUnit,
+		Locality:           in.Locality,
+		Province:           in.Province,
+		SerialNumber:       in.SerialNumber,
```
Comment (amartinezfayo) [inline pkg/server/plugin/credentialcomposer/v1.go:253]: StreetAddress and PostalCode are missing here. We are missing those in the DistinguishedName message in the proto.

File: pkg/server/plugin/credentialcomposer/v1.go [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+		ExtraNames:         extraNames,
+	}, nil
+}
+
+func subjectToV1(in pkix.Name) (*credentialcomposerv1.DistinguishedName, error) {
+	extraNames, err := extraNamesToV1(in.ExtraNames)
+	if err != nil {
+		return nil, err
+	}
+	return &credentialcomposerv1.DistinguishedName{
```
Comment (amartinezfayo) [inline pkg/server/plugin/credentialcomposer/v1.go:264]: Same comment for StreetAddress and PostalCode.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @azdagron, it's looking great!

File: pkg/server/plugin/credentialcomposer/v1_test.go [hunk 4]
```diff
@@ -0,0 +1,941 @@
...
+					Subject: &credentialcomposerv1.DistinguishedName{
+						ExtraNames: []*credentialcomposerv1.AttributeTypeAndValue{
+							{Oid: "NOT AN OID"},
+						},
+					},
+				},
+			},
+			expectCode:    codes.Internal,
+			expectMessage: `credentialcomposer(test): plugin returned invalid X509CA attributes: subject: extra names: invalid OID: non-integer part "NOT AN OID"`,
+		},
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/server/plugin/credentialcomposer/v1_test.go:167]: Actually that validation is not needed. The diagnostics if you send an empty string are fine without that check. I'll just remove that code.

File: pkg/server/plugin/credentialcomposer/v1_test.go (hunk 1 above)
Comment (azdagron) [reply pkg/server/plugin/credentialcomposer/v1_test.go:862]: :+1:

File: pkg/server/plugin/credentialcomposer/v1.go (hunk 2 above)
Comment (azdagron) [reply pkg/server/plugin/credentialcomposer/v1.go:253]: That was intentional on my part.... but I suppose we should add them. They are always supportable through the ExtraNames field though.

**Review by amartinezfayo — APPROVED** [review]
🚀

**Pull request merged**